app/src/ActionModal/components/ModalVoteSubstrate.vue

```diff
@@ -0,0 +1,428 @@
+          return (this.lockingPeriod = "28")
+        case `2x`:
+          return (this.lockingPeriod = "56")
+        case `3x`:
+          return (this.lockingPeriod = "112")
+        case `4x`:
+          return (this.lockingPeriod = "224")
+        case `5x`:
+          return (this.lockingPeriod = "448")
+        case `6x`:
+          return (this.lockingPeriod = "896")
+        case `Set Max`:
+          return (this.lockingPeriod = "896")
+        default:
+          return (this.lockingPeriod = "0")
+      }
+    },
+    totalVotingController() {
+      this.totalVotingPower = (
+        this.lockedBalance * this.selectedLockingOption.multiplier
+      ).toFixed(6)
+    },
+  },
+  apollo: {
+    balances: {
+      query: gql`
+        query($networkId: String!, $address: String!) {
+          balancesV2(networkId: $networkId, address: $address) {
+            id
+            denom
+            available
+            total
+          }
+        }
+      `,
+      /* istanbul ignore next */
+      variables() {
+        return {
+          networkId: this.currentNetwork.id,
+          address: this.address,
+        }
+      },
+      /* istanbul ignore next */
+      skip() {
+        return !this.address
+      },
+      update(result) {
+        this.lockedBalance = result.balancesV2.find(
+          ({ denom }) => denom === this.currentNetwork.stakingDenom
+        ).available
+        this.totalVotingPower = this.lockedBalance
+        return result.balancesV2
+      },
+    },
+  },
+}
+</script>
+
+<style scoped>
+.buttons {
+  display: flex;
+  justify-content: right;
+}
+
+.card {
+  border: solid 1px var(--app-fg-hover);
+  padding: 1rem;
+  border-radius: 5px;
+  margin-right: 0.5rem;
+}
+
+.card-title {
+  display: block;
+  font-size: var(--sm);
+  color: var(--input-bc);
+}
+
+.line {
+  border: solid 1px var(--app-fg-hover);
+}
+
+.flex-row {
+  display: flex;
+  justify-content: space-evenly;
+}
+
+.locked-balance {
+  color: var(--app-nav);
+  font-size: var(--h1);
+  width: 7rem;
+}
+
+.locking-area {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin: 1rem;
+}
+
+.locking-options {
+  width: 100%;
+  margin-top: 1rem;
+  border: solid 1px var(--app-fg-hover);
+  border-radius: 1rem;
+  padding: 0.5rem;
+  justify-content: space-evenly;
+}
+
+.locking-option {
+  margin-right: 0.5rem;
+  padding: 0.5rem;
+  cursor: pointer;
+}
+
+.activeLocking {
+  background-color: var(--app-nav);
+  color: var(--app-bg);
+  border-radius: 1rem;
+}
+
+.locking-option:hover {
+  background-color: var(--app-nav);
+  color: var(--app-bg);
+  border-radius: 1rem;
+}
+
+.action-modal-group.vote-options {
+  padding: 1rem 0;
+  display: flex;
+  flex-direction: column;
+  max-width: 75%;
+  margin: 0 auto;
+}
+
+.vote-options button {
+  margin: 0.5rem;
+  height: 4rem;
+  width: 100%;
+}
+
+.vote-options button.active {
+  background: var(--highlight);
+  border-color: var(--highlight);
+}
+</style>
+<style>
```

Comment:
if this isn't scoped it will hack other components in unexpected ways 🤔

Reply:
yeah, that is why I used id here. This CSS is only applied when the component is present right? Otherwise I can just use ids, actually, and scope it all

Reply:
ah, but there is a problem. The classes are coming from TmField and so on. I need to do that in order to change those classes coming from other components. That is the issue here. How else can you do that? :shrug:  I think there is no other way

Reply:
do you have to use that component?

Reply:
We can use a normal input :shrug: